keanu-project/src/test/java/io/improbable/keanu/tensor/TensorMatchers.java

```diff
@@ -90,6 +91,10 @@ public void describeTo(Description description) {
         };
     }
 
+    public static <T> Matcher<Tensor<T>> tensorEqualTo(Tensor<T> tensor) {
```

Comment:
I would favour overriding `Tensor#equals` rather than add this matcher. Otherwise the user has to remember the different semantics of `tensorEqualTo` vs `equalTo`.

Would like to put that idea to the rest of the team though, as changing the semantics of `Tensor#equals` might have implications.

Reply:
Is there a way to use `equalTo` and keep detailed error messages about diffs? The reason I used the tensor matchers here is because it will give you a detailed error message about how they differ (whether it's the shape or a value, in which index is the value different).

Reply:
If you use `equalTo` then it displays an error message that includes the object's `toString`. So I'd suggest that the better solution here is to write a better `Tensor#toString`.

Reply:
But that won't tell you in which index the value is different.

Reply:
I just noticed that I already wrote a `TensorMatchers#elementwiseEqualTo` but it's not being used anywhere. It checks the values but not the shape.

Reply:
You'll notice that this matcher uses `elementWiseEqualTo` under the hood